src/services/transaction.rs

```diff
@@ -1,16 +1,16 @@
 use uuid::Uuid;
 
-use crate::database;
-use crate::models::transaction::{self, Transaction};
+use crate::{
+    database,
+    models::transaction::{self, Transaction},
+};
 
 // TODO: Use the same connection pool through multiple requests
 
-fn create_transaction(t: Transaction) -> Transaction {
-    let mut conn = database::establish_connection();
-    transaction::create_transaction(&mut conn, t)
+fn create_transaction(conn: &database::DbPool, t: Transaction) -> Transaction {
+    transaction::create_transaction(conn, t)
 }
 
-pub fn list_user_transactions(user_id: Uuid) -> Vec<Transaction> {
-    let mut conn = database::establish_connection();
-    transaction::list_user_transactions(&mut conn, user_id)
+pub fn list_user_transactions(conn: &database::DbPool, user_id: Uuid) -> Vec<Transaction> {
```

Comment:
Just to comment that in the future we will want to depend on types/traits instead of a concrete DbPool implementation. The service layer tends to be quite abstract and very very thin on operations with a simple business logic. 
For example: Every service function would receive a repo as a parameter or use a trait to have this "injection" behaviour. 
BTW I'm ok with the current implementation too.

Reply:
ok